lib/altsvc.c

```diff
@@ -409,7 +409,7 @@ static CURLcode getalnum(const char **ptr, char *alpnbuf, size_t buflen)
   protop = p;
   while(*p && !ISBLANK(*p) && (*p != ';') && (*p != '='))
     p++;
-  len = p - protop;
+  len = (size_t)(p - protop);
```

Comment:
I think this kind of typecast is perhaps the biggest sign that we should not use this warning option. The diff between two `char *` should be possible to store in a `size_t` without a typecast.

Or put another way: do these typecasts really help? They will make writing new code a little more awkward.

Reply:
It pinpoints all the places where the result might be negative,
but the code assumes positive. For the `strchr()` pattern and manual
versions of it, this isn't really useful, but may be in the rest.

I assumed that all existing code is correct, but this isn't readily obvious
while reviewing the ~450 cases.

Reply:
> It pinpoints all the places where the result might be negative, but the code assumes positive

The code does not *assume* this, it knows. The logic is quite easy to follow.

In this and similar cases, this typecast does not add anything at all to a reader. In fact, it is mostly confusing to me since a pointer-delta should already be possible to store in a `size_t' just fine. Should it not?

Have you detected and corrected any actual errors with this exercise?

Reply:
I did some, that were obvious and low-hanging, yes.

But, I try not to touch the actual logic, first because it'd take a very long time to explore each rabbit hole, and second it'd make the patch confusing IMO. Also, it's highly likely the code is correct unless proven otherwise.

What I find that it's difficult to deduct from local contexts that the code is handling the implicit sign conversions conciously.

Am I mistaken that e.g. a 100 - 1000 pointer delta would result in an underflow and make the result a very high number when handled as an unsigned `size_t`?

The other thing driving this is that this is one the few warnings actually recommended by OpenSSF. Assuming they know what they are saying, this might actually be useful, though if course there is nothing saying we must follow their recommendations.